packages/ws/src/ws/WebSocketManager.ts

```diff
@@ -65,6 +65,8 @@ export interface RequiredWebSocketManagerOptions {
 	intents: GatewayIntentBits | 0;
 	/**
 	 * The REST instance to use for fetching gateway information
+	 *
+	 * @deprecated In the future, providing a rest instance will not be required. Provide the `fetchGatewayInformation` function instead.
```

Comment:
Doesn't make much sense to deprecate a required property.
Maybe make it optional and validate in the constructor (one of `rest` or `fetchGatewayInformation` must be provided)